src/nupic/utils/LoggingException.cpp

```diff
@@ -34,7 +34,11 @@ LoggingException::~LoggingException() throw()
   if (!alreadyLogged_) {
     // Let LogItem do the work for us. This code is a bit complex
     // because LogItem was designed to be used from a logging macro
+#if defined(NTA_OS_WINDOWS) && defined(NTA_COMPILER_MSVC) && defined(NDEBUG)
+    auto li = new LogItem(filename_.c_str(), lineno_, LogItem::info);
```

Comment:
Interesting. So I think AppVeyor must be considering a command to have failed if it writes to stderr, regardless of the exit code. Perhaps even if the tests pass, they sometimes write directly to stderr. Our existing tooling only cares about the exit code of the command. If my hypothesis is correct, there are a couple options:

- You could try redirecting stderr to a file, which would hopefully prevent the failure but you would need to make the file a build artifact for debugging. This is the simpler option but probably not the "right" solution, although certainly fine for now.
- You could try eliminating stderr when there isn't a problem. That means leaving LogItem::error in the LoggingException class but figuring out how it is being called during the build.

One way to track this down would be to redirect stderr to a file locally and see what gets written to the file. An even better way to debug would be to put a debug trace in LoggingException to see when it is getting called.